### PR TITLE
10-Registaration Form-Saghar Hosseinmardi

### LINKED ISSUE #10: Registration Form
| name  | about | title  | labels | assignees  |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| "👥 User story"  | Create a new user story  | FORM | REGISTER /TRACKING FORM | TBC  | 

## Story: 
*As a ... user/trainee,
I want to .... fill out a form with information about my team
So that .... my teams activity and PRs are tracked to make it easy to have information about our PR.

## Acceptance Criteria:

1. GIVEN ... I’m a trainee about to start my final project,
- WHEN ... I access the app,
- THEN ... I should be able to fill out a form with mine and my team/group members details, so that our progress can begin to be tracked by the app.

## Checklist:

- [ ] I have defined the user story
- [ ] I have defined some acceptance criteria
- [ ] I have included any implementation notes
- [ ] I have included all relevant links

### PR DESCRIPTION
**Pull request by SagharHosseinmardi**
## Naming Rules

#10-REGISTRATION FORM-SAGHAR HOSSEINMARDI

## Description

1. Added dependencies.
2. Rename index.js to server.js.
3. Created the registration form.

## Related to #10 

Fixes #10 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have carefully reviewed my own code
- [x] I have commented my code
- [x] I have updated any documentation

### CODE REVIEW

File: client/src/components/Form/Form.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+  const [role1, setRole1] = useState("");
+  const [role2, setRole2] = useState("");
+  const [role3, setRole3] = useState("");
+  const [role4, setRole4] = useState("");
+
+  const [githubUsername1, setGithubUsername1] = useState("");
+  const [githubUsername2, setGithubUsername2] = useState("");
+  const [githubUsername3, setGithubUsername3] = useState("");
+  const [githubUsername4, setGithubUsername4] = useState("");
```
Comment (AndrianaOS) [inline client/src/components/Form/Form.js]: We discussed putting lines 4 to 24 in a single useState already. As we said this is ok for now, we will refactor the code as we go along

File: client/src/components/Form/Form.js [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    try {
+      const response = await fetch("http://localhost:8000/submit-form", {
```
Comment (AndrianaOS) [inline client/src/components/Form/Form.js]: Change the fetch url to the render url for the backend so we can all run the code as we all dont have the local environment this code runs from

File: client/src/components/Form/Form.js [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+          role4,
+          githubUsername1,
+          githubUsername2,
+          githubUsername3,
+          githubUsername4,
+        }),
+      });
+
+      const data = await response.json();
+      console.log(data);
```
Comment (AndrianaOS) [inline client/src/components/Form/Form.js]: Remember to take this out so its not console logging information every time it runs

File: client/src/components/Form/Form.js [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+          id="teamName"
+          name="teamName"
+          value={teamName}
+          onChange={(e) => setTeamName(e.target.value)}
+          required
+        />
+      </div>
+      <div>
+        <label htmlFor="projectBrief">Project Brief:</label>
+        <textarea
```
Comment (AndrianaOS) [inline client/src/components/Form/Form.js]: you can change this `textarea` into an input as it will only be accepting the url of the project brief

**Review by AndrianaOS — CHANGES_REQUESTED** [review]
Ive made some comments on some little changes. You can ignore the useState comments for now as the code works and refactor the code later on but change the fetch url to the render backend url, remove the console.logs and change the text area in the project brief section into an input instead.
Once that is done, it can be merged

File: client/src/components/Form/Form.js (hunk 1 above)
Comment (SagharHosseinmardi) [reply client/src/components/Form/Form.js]: This part has been refactored.

**Review by AndrianaOS — APPROVED** [review]
All good. Thank you for the changes Saghar

**Review by AndrianaOS — APPROVED** [review]

**Pull request merged**